### ISSUE TITLE
Bad/incomplete example for Understanding 3.3.2

### ISSUE THREAD

**Issue by patrickhlauke** [w3c/wcag#755]
In https://www.w3.org/TR/UNDERSTANDING-WCAG20/minimize-error-cues.html the last example:

> A U.S. phone number separates the area code, exchange, and number into three fields. Parentheses surround the area code field, and a dash separates the exchange and number fields. While the punctuation provides visual clues to those familiar with the U.S. telephone number format, the punctuation is not sufficient to label the fields. The single "Phone number" label also cannot label all three fields. To address this, the three fields are grouped in a fieldset with the legend "Phone number". Visual labels for the fields (beyond the punctuation) cannot be provided in the design, so invisible labels are provided with the "title" attribute to each of the three fields. The value of this attribute for the three fields are, respectively, "Area Code", "Exchange", and "Number".

as the `title` attribute won't show for keyboard users, it's not really a good example...should the example be removed, or at least generalised to say that a tooltip is shown when the field is hovered or focused (leaving it up to the reader to work out how it's done, i.e. as a custom-built tooltip not just `title`) ?

**Comment by awkawk** [comment]
I don't think that the tooltip content is required to be keyboard accessible in this case. The visual information present is sufficient to allow a sighted user to know what to input. The title (or aria-label) is primarily for the AT user.

**Comment by patrickhlauke** [comment]
So a mouse user can see that the three fields are "Area Code", "Exchange", "Number". AT user can hear that the three fields are "Area Code", "Exchange", "Number". But a sighted keyboard user has to know/guess this?

**Comment by awkawk** [comment]
A sighted user gets their context from the visual information (3 separate fields, punctuation, knowing the structure of phone numbers). This would pass if aria-label was used instead of title, right?

**Comment by patrickhlauke** [comment]
i'd say no, it wouldn't either? borderline, but this example seems rather tainted in any case if we're having such discussions here as well...

**Comment by awkawk** [comment]
I do think that it would pass using aria-label. Which SC would it fail? As a non-US person "area code", "exchange", and "number" probably don't provide great clarity. For that matter, I suspect that most US people wouldn't typically use "exchange" so the visual information is the most important.

**Comment by patrickhlauke** [comment]
> Which SC would it fail?

3.3.2?

**Comment by patrickhlauke** [comment]
turning this around: why would you think it passes 3.3.2? are there labels/instructions provided about what the user is meant to enter into the three separate inputs? are the parentheses and dash here supposed to be sufficiently clear? or does it not just rely on assumed knowledge on the part of the user, without making it explicitly clear what the three inputs are meant to represent?

**Comment by awkawk** [comment]
A form that is looking for a phone number that breaks it into three controls with an overall label of "phone number" is sufficiently labeled for sighted users to know what to do. The individual controls still need an accessible name since the visual labelling isn't apparent to visually impaired users.

If I go to a different country I'm more likely to make a mistake whether I am sighted or not because I'm not familiar with the conventions.

**Comment by patrickhlauke** [comment]
>  is sufficiently labeled for sighted users to know what to do

and i guess this is the crux of our disagreement...

> If I go to a different country I'm more likely to make a mistake whether I am sighted or not because I'm not familiar with the conventions.

...but isn't the point of 3.3.2 to fight exactly this problem? am confused by the cavalier attitude here, i must admit.

but ok, assuming you're right then...why suggest `title` here at all? if this example is only meant to fix it for AT users / to provide an accessible name, it should be an example for 4.1.2 not 3.3.2.

in short...let's remove this example as it seems rather edge-casey and confusing/mixing different concerns (or expand it as i said, to make it clear that we're talking about a tooltip rather than `title` per se)

**Comment by awkawk** [comment]
> and i guess this is the crux of our disagreement...

OK. Others can weigh in.

> ...but isn't the point of 3.3.2 to fight exactly this problem? am confused by the cavalier attitude here, i must admit.

There's no cavalier attitude, we're just talking about what is required. If you want to advocate for what is BEST then that's ok, but that's different.

> but ok, assuming you're right then...why suggest title here at all? if this example is only meant to fix it for AT users / to provide an accessible name, it should be an example for 4.1.2 not 3.3.2.

Becasue when this technique was made aria didn't exist.

> in short...let's remove this example as it seems rather edge-casey and confusing/mixing different concerns (or expand it as i said, to make it clear that we're talking about a tooltip rather than title per se)

I don't regard this as a very high priority. You are welcome to suggest specific changes, but there are  many more pressing items for the WG to review.

**Comment by goetsu** [comment]
Regarding current situation, my understanding is as title is currently listed as sufficient technique the current example is compliant. The problem of having it not displayed on focus was at the time more seen as an UAAG problem.

But then we have a more wider problem to know if WG even consider to have label (and not only instructions) visible only on focus as a compliant solution regarding 3.3.2 I would say yes based on my interpretation but others seem to disagree. Then depending on WG answer the understand and associated techniques may need to be clarified on this.

**Comment by mraccess77** [comment]
Generally I would fail form fields that only have labels on focus, however, in other situations I've seen use of color issues that are addressed when an item is focus -- for example a chart with a legend that uses color when the corresponding legend is highlighted when the item in the chart is focused.  So I agree there isn't clear guidance on this topic.

**Comment by frex65** [comment]
Thinking about a different scenario where there are three text fields for entering a date. is the format DD/MM/YYYY or MM/DD/YYYY? If this information is only available via the “title” attribute, mouse and screen reader users are covered but there is a lot of scope for sighted keyboard users to get it wrong.

**Comment by alastc** [comment]
Going back to the original example, how about this tweak (from #101) :

> The single "Phone number" label also cannot label all three fields. To address this, the three fields are grouped in a fieldset with the legend "Phone number". Visual labels for the fields (beyond the punctuation) cannot be provided in the design, and **all three fields are considered to be visually labeled by the legend. To meet other criteria, each field would also need to have an accessible name set, for example by using the title or aria-label attribute.**

**Comment by alastc** [comment]
PR for that change (and a couple of others) in #1205

**Comment by patrickhlauke** [comment]
still think the example itself is rather tainted and US-centric - it pretty much relies on the idea that yeah, everybody knows that the first bit is the area code, the second part is the exchange, the third is the number ... or that they just don't need to know/care.

agree there with @frex65 that if instead the legend just said "Date" and then had 3 inputs...and those 3 inputs then have a `title` that shows for mouse users what each field is ("Day", "Month", "Year") but that this wouldn't be visible to keyboard users, then that would fail...

but yeh...as long as devs then don't get surprised if they still fail 3.3.2 even if they diligently only use a legend and title (as it would then not provide the info for sighted keyboard users), then sure.

**Comment by alastc** [comment]
> it pretty much relies on the idea that yeah, everybody knows that the first bit is the area code, the second part is the exchange, the third is the number ... or that they just don't need to know/care. 

The first part of the example does explain that. I didn't know what each was for either, but the example just explains to everyone that there is a format, and there are indicators but not labels. I think a credit card example would also work (split into 4 fields), but its the same situation where we'd need to explain it.

> but yeh...as long as devs then don't get surprised if they still fail 3.3.2 even if they diligently only use a legend and title (as it would then not provide the info for sighted keyboard users), then sure.

Speaking of failures, we do have [F82](https://www.w3.org/WAI/WCAG21/Techniques/failures/F82). It would break the format to have a failure example, but how about a success?

E.g: A field for entering a date uses three inputs: month, day and year. They are closely spaced with dashes inbetween but there are no visible labels for each input. A legend is used for all three which indicates the format "Date, MM-DD-YYYY".

**Comment by innoticFR** [comment]
I agree that a date would be a more universal example but isn't this last example too close to the second one *"A field for entering a date contains initial text which indicates the correct format for the date."* ?

Maybe using list boxes, we can focus also more on the fact that the elements might be *"unambiguous"* in the context rather than the format:

> A field for entering a date uses three list boxes: month (listed in text), day (from 1 to 31) and year (listed in 4 digits values). Each list box is visually unambiguous to each other but there are no visible labels for each one. The legend *"Birth date"* is used for all three.

Other scenarios implying multiple fields where showing individual labels could be confusing : IBAN (or bank account number), phone (international prefix, local number), or any fixed size serial number / promo code, but those are examples where input masking would be a more suitable solution.

**Comment by patrickhlauke** [comment]
see also https://web-a11y.slack.com/archives/C071U5Y3H/p1782311516518989
